react/features/recording/middleware.js

```diff
@@ -81,15 +85,19 @@ MiddlewareRegistry.register(store => next => action => {
             = getSessionById(store.getState(), action.sessionData.id);
 
         if (updatedSessionData.mode === JitsiRecordingConstants.mode.FILE) {
-            const { OFF, ON } = JitsiRecordingConstants.status;
+            const { PENDING, OFF, ON } = JitsiRecordingConstants.status;
 
             if (updatedSessionData.status === ON
                 && (!oldSessionData || oldSessionData.status !== ON)) {
                 store.dispatch(playSound(RECORDING_ON_SOUND_ID));
+                store.dispatch(hidePendingRecordingMessage());
             } else if (updatedSessionData.status === OFF
                 && (!oldSessionData || oldSessionData.status !== OFF)) {
                 store.dispatch(stopSound(RECORDING_ON_SOUND_ID));
                 store.dispatch(playSound(RECORDING_OFF_SOUND_ID));
+            } else if (updatedSessionData.status === PENDING
+                && (!oldSessionData || oldSessionData.status !== PENDING)) {
+                store.dispatch(showPendingRecordingMessage());
```

Comment:
I think it'd be good actually to take into account error states right now too. A recording can go from on to error. I'm not sure if I ever tested pending to error. If a recording fails to start, there is no recording session so the error is handle elsewhere.